src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -50,9 +50,11 @@ internal class ConnectionManager : IConnectionManager {
 
             _disposableBag = DisposableBag.Create<ConnectionManager>()
                 .Add(_statusBarViewModel)
-                .Add(() => _sessionProvider.BrokerStateChanged -= BrokerStateChanged);
+                .Add(() => interactiveWorkflow.RSession.Connected -= RSession_Connected)
+                .Add(() => interactiveWorkflow.RSession.Disconnected -= RSession_Disconnected);
 
-            _sessionProvider.BrokerStateChanged += BrokerStateChanged;
+            interactiveWorkflow.RSession.Connected += RSession_Connected;
```

Comment:
The problem is that indication is confusing. It shows everything green but host is down. However, green can be treated as 'connected' only. I am wondering if user will get confused why things are not working especially when connected to remote.